core/job/job.go

```diff
@@ -560,3 +560,45 @@ type DeployState string
 func (d DeployState) String() string {
 	return string(d)
 }
+
+type DownstreamWithHierarchy struct {
```

Comment:
can we not to create another struct for managing downstream? if the use case only for showing the information about the downstream to be deleted

Reply:
I was planning to encode this level information inside the existing Downstream struct, but this field usage would be only on this change's specific functionality, so decided to create a new struct.

other than refetching only the direct dependencies, any suggestions mas?

Reply:
updated: use list of downstream list approach